Mindshock/mindshield modifications

Rhials opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Reduces the length of the brainwashing from six minutes to two. Adds a second charge to the mindshock gland to compensate.

Makes it so that having a mindshield will protect you from the effects of a mindshock gland. The includes the passive effects that mindshock glands have on people near them, as well as the brainwash command from the mental interface device.

Since the implant itself is handling the brainwash broadcast, it will still succeed in broadcasting the message even if the gland holder is mindshielded.  Mindshields only prevent you from _receiving_ the brainwash message. Tinfoil hat functionality is untouched and remains the only way to block the message from being broadcast.

Very open to feedback on this one, please let me know what you think.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

The entire purpose of mindshields are to protect from brainwashing. If it can shield you from mutineer flashes, traitor hypnoflashes, and the eldritch influence of Nar'Sie herself, it should be able to protect you from alien brainwashing. You already have very little way of seeing it coming, and it can control an entire room of people (like a packed medbay trying to remove abductor glands), so there should be some form of counterplay if the crew knows what the abductors are up to.

As for the length of the hypnosis being shortened, six minutes really starts to drag on when you're at the receiving end of the brainwashing. Abductors still have brainwash surgery for their long-term bits, and with the second charge the mindshock brainwash is a bit more flexible in its short-term comedy potential.

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
balance: The mindshock gland mind control has been changed from a single, six-minute command to two, two-minute long commands. Also, those who are mindshielded will resist the brainwashing effects and the brain-melting waves the gland passively emits.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rhials committed:
makes the mindshock gland not too busted

mind sheild :)

Rhials committed:
now with feedback!

wowee

Rhials committed:
i'm not using the word "moment" twice in one PR

Rhials committed:
longer

longer

Rhials committed:
two

two

MrMelbert commented on code/modules/antagonists/abductor/equipment/glands/mindshock.dm:
```diff
@@ -4,8 +4,8 @@
-	mind_control_uses = 1
-	mind_control_duration = 6000
+	mind_control_uses = 2
+	mind_control_duration = 1200
```
You should put this in `seconds` or `minutes`. 
`120 SECONDS` or `2 MINUTES`.

MrMelbert commented on code/modules/antagonists/abductor/equipment/glands/mindshock.dm:
```diff
@@ -15,16 +15,19 @@
-				H.adjustOrganLoss(ORGAN_SLOT_BRAIN, 10, 160)
-			if(3)
-				H.hallucination += 60
+		if(!HAS_TRAIT(H, TRAIT_MINDSHIELD))
+			switch(pick(1,3))
+				if(1)
+					to_chat(H, span_userdanger("You hear a loud buzz in your head, silencing your thoughts!"))
+					H.Stun(50)
+				if(2)
+					to_chat(H, span_warning("You hear an annoying buzz in your head."))
+					H.add_confusion(15)
+					H.adjustOrganLoss(ORGAN_SLOT_BRAIN, 10, 160)
+				if(3)
+					H.hallucination += 60
+		else
+			to_chat(H, span_notice("You hear a faint hum fill your ears, which quickly dies down."))
```
You can use an early continue here. 
```
if(HAS_TRAIT(H, TRAIT_MINDSHIELD))
	to_chat(H, span_notice("You hear a faint hum fill your ears, which quickly dies down."))
	continue

...
```

MrMelbert commented on code/modules/antagonists/abductor/equipment/glands/mindshock.dm:
```diff
@@ -37,15 +40,18 @@
+			broadcasted_mobs += H
+			to_chat(H, span_userdanger("You suddenly feel an irresistible compulsion to follow an order..."))
+			to_chat(H, span_mind_control("[command]"))
 
-		message_admins("[key_name(user)] broadcasted an abductor mind control message from [key_name(owner)] to [key_name(H)]: [command]")
-		log_game("[key_name(user)] broadcasted an abductor mind control message from [key_name(owner)] to [key_name(H)]: [command]")
+			message_admins("[key_name(user)] broadcasted an abductor mind control message from [key_name(owner)] to [key_name(H)]: [command]")
+			log_game("[key_name(user)] broadcasted an abductor mind control message from [key_name(owner)] to [key_name(H)]: [command]")
 
-		var/atom/movable/screen/alert/mind_control/mind_alert = H.throw_alert(ALERT_MIND_CONTROL, /atom/movable/screen/alert/mind_control)
-		mind_alert.command = command
+			var/atom/movable/screen/alert/mind_control/mind_alert = H.throw_alert(ALERT_MIND_CONTROL, /atom/movable/screen/alert/mind_control)
+			mind_alert.command = command
+		else
+			to_chat(H, span_notice("You hear a low drone as something foreign attempts to enter your mind, but the noise fades after a few moments."))
```
And you can do an early continue here as well. 
```
if(HAS_TRAIT(H, TRAIT_MINDSHIELD))
	to_chat(H, span_notice("You hear a low drone as something foreign attempts to enter your mind, but the noise fades after a few moments."))
	continue
	
...
```

Rhials committed:
pretty code :)

SkeletalElite commented:
Abductors can just remove your mindshield though, right? Rather than making them do more surgery, maybe just make the mind shield implant break?

ATH1909 commented:
> Abductors can just remove your mindshield though, right? Rather than making them do more surgery, maybe just make the mind shield implant break?

that'll give your abductee nature away, though

tralezab commented on code/modules/antagonists/abductor/equipment/glands/mindshock.dm:
```diff
@@ -15,6 +15,10 @@
 	for(var/mob/living/carbon/H in orange(4,T))
```
if we use the variable, we should rename bad vars while we're here. change H (and places that use it) to something better!

tralezab reviewed (changes requested):
balance wise I think this is fine enough to make, mindshield should at least work intuitively. I remember historically the rename from loyalty to mindshield was done for player intuition towards what the implant actually does.

codewise just a little housekeeping

Rhials committed:
h = m++;

Rhials committed:
oops

sorry i was watching a jerma vod while i was coding this and missed something sorry sorry oops
